src/app/calculator/utilities/natural-gas-reduction/natural-gas-reduction-form/natural-gas-reduction-form.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Settings } from '../../../../shared/models/settings';
+import { FormGroup } from '@angular/forms';
+import { OuterSubscriber } from 'rxjs/OuterSubscriber';
+import { NaturalGasReductionResults, NaturalGasReductionService } from '../natural-gas-reduction.service';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'app-natural-gas-reduction-form',
+  templateUrl: './natural-gas-reduction-form.component.html',
+  styleUrls: ['./natural-gas-reduction-form.component.css']
+})
+export class NaturalGasReductionFormComponent implements OnInit {
+  @Input()
+  settings: Settings;
+  @Input()
+  form: FormGroup;
+  @Input()
+  index: number;
+  @Input()
+  isBaseline: boolean;
+  @Output('emitCalculate')
+  emitCalculate = new EventEmitter<{ form: FormGroup, index: number, isBaseline: boolean }>();
+  @Output('emitChangeField')
+  emitChangeField = new EventEmitter<string>();
+
+  measurementOptions: Array<{ value: number, name: string }>;
+  idString: string;
+  individualResults: NaturalGasReductionResults;
+  format: any;
+
+  constructor(private naturalGasReductionService: NaturalGasReductionService) { }
+
+  ngOnInit() {
+    this.getFormat();
+    if (this.isBaseline) {
+      this.idString = this.index.toString();
+    }
+    else {
+      this.idString = 'modification_' + this.index;
+    }
+    this.initMeasurementOptions();
+    this.calculate();
+  }
+
+  initMeasurementOptions() {
```

Comment:
If this array never changes, it would make sense to just initialize it with these values. Eliminates the step of calling, creating and pushing the array data on init.
//can be done on line 27
measurementOptions = [
 {value: 0, name: 'Flow Meter Method'},
 ...
]